lib/core/viewmodels/more_viewmodel.dart

```diff
@@ -66,85 +81,37 @@ class MoreViewModel extends FutureViewModel {
     } on Exception catch (e) {
       onError(e);
     }
-    UserRepository().logOut();
+    await _preferencesService.clear();
+    await _userRepository.logOut();
+
+    settingsManager.resetLanguageAndThemeMode();
+    _courseRepository.sessions.clear();
+    _courseRepository.coursesActivities.clear();
     setBusy(false);
     // Dismiss alertDialog
     navigationService.pop();
     navigationService.pushNamedAndRemoveUntil(RouterPaths.login);
-    showToast(_appIntl.login_msg_logout_success);
+    settingsManager.notifyListeners();
```

Comment:
```suggestion
    notifyListeners();
```
We are in a viewmodel which is based on ChangeNotifier. Is there a reason for calling the notifyListeners of settingsManager?

Reply:
I think it's to make sure the listeners of settings_manager aware that the language and the theme has been reset (so it fall back on system preference when login out). I'm not sure if we need to call the notifyListeners on the ViewModel, I try the two ways and everything works nonetheless. 
Perhaps, an idea to make it cleaner, would be to put the call to `settingsManager.notifyListeners` inside the function `settingsManager.resetLanguageAndThemeMode` so the logout doesn't know it's calling the notify.

What do you think @AlexandreMarkus ?

Reply:
I think you would have to test it and see if it still does the same behavior